swap_meet/vendor.py

```diff
@@ -1,2 +1,64 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None) :
+        if not inventory:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self,item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+    def get_by_category(self, category):
+        item_list = []
+        for item in self.inventory:
+            if category == item.category:
+                item_list.append(item)
+
+        return item_list
+
+    def swap_items(self, vendor: 'Vendor', my_item, their_item):
+        if my_item not in self.inventory:
+            return False
+        if their_item not in vendor.inventory:
+            return False
+        self.remove(my_item)
+        vendor.add(my_item)
+        vendor.remove(their_item)
+        self.add(their_item)
+        return True
+
+    def swap_first_item(self, vendor: 'Vendor'):
+        if not self.inventory or not vendor.inventory:
+            return False
+
+        my_item = self.inventory[0]
+        their_item = vendor.inventory[0]
+
+        return self.swap_items(vendor, my_item, their_item)
+        
+    def get_best_by_category(self, category = ''):
+        if not self.inventory:
+            return None
+
+        category_items = []
+        for item in self.inventory:
+            if item.category == category:
+                category_items.append(item)
```

Comment:
Consider reusing your get_by_category method that you already defined so you don't have to loop and add to a list again.